### DISCUSSION TITLE
is it possible to get "moved" coloring support outside of git?

### DISCUSSION THREAD

**Discussion by Dieterbe** [dandavison/delta#964] [General]
Hello, when invoking `delta a.txt b.txt` on 2 files, ie without using git, is it possible to also get the special highlighting for moved content? thanks!
the tool looks great btw :-)

**Comment by dandavison** [comment]
Good call! You're right, it's a bug, so I've created an issue from this discussion: https://github.com/dandavison/delta/issues/965. I just made a release yesterday so I'll include this fix if I have to make a serious bug fix release, but otherwise, to get the fix, you could [build delta from the git repo](https://dandavison.github.io/delta/build-delta-from-source.html) or if using homebrew install using `brew install --HEAD git-delta`.

**Comment by dandavison** [comment]
Bug fixed in https://github.com/dandavison/delta/pull/966

**Comment by Dieterbe** [comment]
my mind is kind of blown. i assumed that:
1) 'move detection' is a git specific feature
2) implementing 'move detection' in regular (non-git) diff mode, would be a huge project, and likely out of scope for delta.

So, great news that this is fixed in git. thanks!

**Comment by dandavison** [comment]
Haha, no, you probably looked at the fix -- much easier than that :) `delta a b` calls `git diff --no-index a b` under the hood; the bug was just that we weren't passing `--color` to `git diff`, since the way that delta supports color-moved is by parsing the ANSI color escapes that git emits.

**Comment by Dieterbe** [comment]
oh. so `git diff --no-index` is basically an alternative for `/usr/bin/diff` .. very interesting.

**Reply by dandavison** [reply]
Yes, that's right. You can look at https://github.com/dandavison/delta/issues/543 if you're interested. (There's one limitation currently which is that it doesn't work with process substitution; but see discussion below https://github.com/dandavison/delta/issues/929#issuecomment-1020597339 where a workaround is discussed but not implemented yet.)

**Reply by Dieterbe** [reply]
nice nice. cheers Dan!

**Comment by fakhrulhilal** [comment]
Sorry, I don't understand. Does it mean it's possible to use coloring outside git command? And where the configuration will be taken from, is it the same .gitconfig file or do we use another config file? 
I just realize it's bug. But, definitely it's great feature.

**Reply by dandavison** [reply]
Yes, you can do `delta fileA fileB` and it will show a colored diff with all delta features. It takes the config from ~/.gitconfig as usual.
